src/modifier/modifier.ts

```diff
@@ -138,11 +138,11 @@ export abstract class Modifier {
     return false;
   }
 
-  shouldApply(_args: any[]): boolean {
+  shouldApply(...args: Parameters<this["apply"]>): boolean {
     return true;
   }
 
-  abstract apply(args: any[]): boolean | Promise<boolean>;
+  abstract apply(...args: unknown[]): boolean | Promise<boolean>;
```

Comment:
Something that just came to my mind. Why didn't we consider this too?
```suggestion
  abstract apply<T extends Array = unknown[]>y(...args: T): boolean | Promise<boolean>;
```
?

Reply:
This prevents spreading. By forcing `...args: T` the parameters of the apply function of all children must be an array.